tests/v2/validation/charts/istio_test.go

```diff
@@ -0,0 +1,293 @@
+
+	i.client = client
+
+	//Get clusterName from config yaml
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(i.T(), clusterName, "Cluster name to install is not set")
+
+	//Get clusterID with clusterName
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(i.T(), err)
+
+	//Change kiali and jaeger paths if it's not local cluster
+	if clusterID != clusterName {
+		kialiPath = fmt.Sprintf("k8s/clusters/%s/%s", clusterID, kialiPath)
+		tracingPath = fmt.Sprintf("k8s/clusters/%s/%s", clusterID, tracingPath)
+	}
+
+	//Get latest versions of monitoring & istio charts
+	latestIstioVersion, err := client.Catalog.GetLatestChartVersion(charts.RancherIstioName)
+	require.NoError(i.T(), err)
+	latestMonitoringVersion, err := client.Catalog.GetLatestChartVersion(charts.RancherMonitoringName)
+	require.NoError(i.T(), err)
+
+	//Create project
+	projectConfig := &management.Project{
+		ClusterID: clusterID,
+		Name:      exampleAppProjectName,
+	}
+	createdProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(i.T(), err)
+	require.Equal(i.T(), createdProject.Name, exampleAppProjectName)
+	i.project = createdProject
+
+	i.chartInstallOptions = &chartInstallOptions{
+		monitoring: &charts.InstallOptions{
+			ClusterName: clusterName,
+			ClusterID:   clusterID,
+			Version:     latestMonitoringVersion,
+			ProjectID:   createdProject.ID,
+		},
+		istio: &charts.InstallOptions{
+			ClusterName: clusterName,
+			ClusterID:   clusterID,
+			Version:     latestIstioVersion,
+			ProjectID:   createdProject.ID,
+		},
+	}
+}
+
+func (i *IstioTestSuite) TestIstioChart() {
+	subSession := i.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := i.client.WithSession(subSession)
+	require.NoError(i.T(), err)
+
+	monitoringFeatureOptions := &charts.RancherMonitoringOpts{
+		IngressNginx:         true,
+		RKEControllerManager: true,
+		RKEEtcd:              true,
+		RKEProxy:             true,
+		RKEScheduler:         true,
+	}
+	i.T().Log("Installing monitoring chart with the latest version")
+	err = charts.InstallRancherMonitoringChart(client, i.chartInstallOptions.monitoring, monitoringFeatureOptions)
+	require.NoError(i.T(), err)
+
+	i.T().Log("Waiting monitoring chart deployments to have at least one available replicas")
+	err = charts.WatchAndWaitDeployments(client, i.project.ClusterID, charts.RancherMonitoringNamespace)
+	require.NoError(i.T(), err)
+
+	istioFeatureOptions := &charts.RancherIstioOpts{
+		IngressGateways: true,
+		EgressGateways:  false,
+		Pilot:           true,
+		Telemetry:       true,
+		Kiali:           true,
+		Tracing:         true,
+		CNI:             false,
+	}
+	i.T().Log("Installing istio chart with the latest version")
+	err = charts.InstallRancherIstioChart(client, i.chartInstallOptions.istio, istioFeatureOptions)
+	require.NoError(i.T(), err)
+
+	i.T().Log("Waiting istio chart deployments to have at least one available replicas")
```

Comment:
Wouldn't it make sense to wait for the expected number of replicas here? 
That way we make sure all workers nodes are working in the cluster if it's using a DaemonSet